proposals/base-relative-assets.md

```diff
@@ -0,0 +1,132 @@
+
+- /foo/
+- /foo/bar/
+- /foo/bar/baz/
+
+If we statically generate relative links then these will be incorrect. So we will throw an error if using a 404.astro page with `'relative'` and suggest using `'absolute'` instead.
+
+## Imported images
+
+Images that are imported will *not* be relative. That means this:
+
+```astro
+---
+import penguinUrl from '../assets/penguin.png';
+---
+
+<img src={penguinUrl} />
+```
+
+It might be *possible* to make this work through a Vite plugin but would be a bit of work and might be error prone. 
+
+# Testing strategy
+
+Given that resolving URLs will be done in a few places, it makes sense to create a module within `src/core` for this algorithm. This module will be unit testable for all of the scenarios. We should probably also move the runtime resolver (which is used internally by the runtime for producing `<astro-island>` URLs) since it is shared logic between SSG and SSR that is related.
+
+Additionally some fixture tests will be included for the matrix of scenarios; SSG and SSR mode, `absolute` and `relative` options, as well as the types of assets; stylesheets, hoisted scripts, and astro islands.
+
+# Drawbacks
+
+- This adds some complexity to the codebase in the usual way that adding a config option does.
+  - However this is also an opportunity to clean up the code and have just 1 algorithm for determing how a URL gets resolved that we create.
+- This will not fix the problem of `<img>`s you create. this is especially important in Markdown files, where manually adding relative links is required. So people might find this solution inadequate since it doesn't fix all of your problems.
+  - The counter point is that you can use `<base>` tags and then all of your img links can just be bare.
+
+# Alternatives
+
+The main alternative I've considered here is instead of `relative`, having a `bare` or `base` option. That would add a `<base>` tag to your page and then use bare links like `<link rel="stylesheet" href="assets/main.hash.css">`. 
+
+The main reason why I did not proceed with this idea is because we *cannot* use bare specifiers for module imports. Those are required by browsers to either be absolute or relative (or use an importmap, which isn't what we want here).
+
+# Adoption strategy
+
+Please consider:
+
+- Given that this is a common support topic we would want to create documentation around this new feature and probably some sort of guide on how to create Astro pages that use base relative links.
+- `absolute` still makes sense as a sane default. Given the 404 limation of `relative` I don't think we would ever make it the default.
+
+# Unresolved questions
+
+It's unclear to me how/if the Image component in `@astrojs/image` can build relative URLs. It's possible that it *can*, given that it knows the current page URL. That's something to be explored as supporting relative links there would be a good answer for the `<img>` limitation.
```

Comment:
Yes, I was also thinking I’d expect the image integration to be compatible with this config setting.

Could be a good thing to bear in mind so that the tools needed to build relative paths are available to integrations as well as internally.